src/main/antlr4/Expr.g4

```diff
@@ -0,0 +1,65 @@
+grammar Expr;
+
+@header {
+package io.mfj.expr.antlr4;
+}
+
+root : expression EOF;
+
+expression: term ( CONJUNCTION expression )?;
+
+term : statement | parens | not;
+
+parens: '(' expression ')';
+
+not : notStart expression ')';
+notStart : NOTP | notSpaceP;
+fragment NOTP : [nN][oO][tT]'(';
+notSpaceP : NOT '(';
+NOT: [nN][oO][tT];
+
+statement : value OPERATOR value ( OPERATOR value )?;
+
+value : value MATH_OP value | varName | literalValue | list;
+
+CONJUNCTION: AND | OR;
+fragment AND : [aA][nN][dD];
+fragment OR : [oO][rR];
+
+OPERATOR : '!=' | '<>' | '>=' | '<=' | '>' | '<' | '=~' | '=' | IN | CONTAINS ;
+
+fragment IN : '!'?[iI][nN];
+fragment CONTAINS: '!'?[cC][oO][nN][tT][aA][iI][nN][sS];
+
+literalValue : nul | number | string | bool | regex | date | time | datetime;
+
+nul: 'null';
```

Comment:
The inconsistent case-sensitivity (e.g. if I'm reading correctly, `null` is case-sensitive while `not` is not) is surprising to me. Is that matching previous behavior? IIRC, the parboiled parser was completely case-insensitive for language keywords; but even if it wasn't, it seems like we probably should be either completely case-sensitive or completely case-insensitive, not a mix depending on the specific keyword.